File: PHP/design_hashset.php
<?php
class MyHashSet
{
    /**
     * @param Integer[] $hashSet
     */
    public $hashSet;
    function __construct() {
        $size = 1000001;
        $this->hashSet = array_fill(0, $size, -1);
    }

    /**
     * @param Integer $key
     * @return NULL
     */
    function add($key) {
        if (!$this->contains($key)) {
            $this->hashSet[$key] = $key;
        }
    }

    /**
     * @param Integer $key
     * @return NULL
     */
    function remove($key) {
        if ($this->contains($key)) {
            $this->hashSet[$key] = -1;
        }
    }

    /**
     * @param Integer $key
     * @return Boolean
     */
    function contains($key) {
        return $this->hashSet[$key] !== -1;
    }
}

/**
 * Your MyHashSet object will be instantiated and called as such:
 * $obj = MyHashSet();
 * $obj->add($key);
 * $obj->remove($key);
 * $ret_3 = $obj->contains($key);
 */

File: Go/design_hashset.go
package main

type MyHashSet struct {
	bucket []int
}

func Constructor() MyHashSet {
	size := 1_000_001
	bucket := make([]int, size)
	for i := range bucket {
		bucket[i] = -1
	}
	return MyHashSet{
		bucket: bucket,
	}
}

func (this *MyHashSet) Add(key int) {
	hash := key % len(this.bucket)
	if key != this.bucket[hash] {
		this.bucket[hash] = key
	}
}

func (this *MyHashSet) Remove(key int) {
	hash := key % len(this.bucket)
	if key == this.bucket[hash] {
		this.bucket[hash] = -1
	}
}

func (this *MyHashSet) Contains(key int) bool {
	hash := key % len(this.bucket)
	return this.bucket[hash] == key
}
